tests/Tests/500__jupyterhub/test-jupyterlab-git.robot

```diff
@@ -0,0 +1,176 @@
+*** Settings ***
+Resource            ../../Resources/ODS.robot
+Resource            ../../Resources/Common.robot
+Resource            ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource            ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Library             OpenShiftCLI
+Library             DebugLibrary
+
+Suite Setup         Server Setup
+Suite Teardown      End Web Test
+
+
+*** Variables ***
+${REPO_URL}         ****
+${DIR_NAME}         Python
+${FILE_PATH}        Python/file.ipynb
+${COMMIT_MSG}       commit msg2
+
+
+*** Test Cases ***
+Verify Pushing Project Changes Remote Repository
+    [Documentation]    Verifies that changes has been pushed successfully to remote repository
+    [Tags]    ODS-326    Sanity    Tier1
+    Set Staging Status
+    ${randnum}=    Generate Random String    9    [NUMBERS]
+    ${commit_message}=    Catenate    ${COMMIT_MSG}    ${randnum}
+    Push Some Changes to Repo
+    ...    ${GITHUB_USER.USERNAME}
+    ...    ${GITHUB_USER.TOKEN}
+    ...    ${FILE_PATH}
+    ...    ${REPO_URL}
+    ...    ${commit_message}
+    Clean Up Server
+
+Verify Updating Project With Changes From Git Repository
+    [Documentation]    Verifies that changes has been pulled successfully to local repository
+    [Tags]    ODS-324    Sanity    Tier1
+    Set Staging Status
+    Clone Git Repository And Open    ${REPO_URL}    ${FILE_PATH}
+    Sleep    1s
+    Open New Notebook
+    ${commit_msg1}=    Get Last Commit Message
+    Add and Run JupyterLab Code Cell in Active Notebook    ! mkdir ../folder/
+    Sleep    2s
+    Open Folder or File    folder
+
+    ${randnum}=    Generate Random String    9    [NUMBERS]
+    ${commit_message}=    Catenate    ${COMMIT_MSG}    ${randnum}
+    #now do here some changes
+    Push Some Changes to Repo
+    ...    ${GITHUB_USER.USERNAME}
+    ...    ${GITHUB_USER.TOKEN}
+    ...    folder/${FILE_PATH}
+    ...    ${REPO_URL}
+    ...    ${commit_message}
+
+    #go to previous dir
+    Close All JupyterLab Tabs
+
+    Open Folder or File    ${DIR_NAME}
+
+    Open With JupyterLab Menu    Git    Pull from Remote
+    Sleep    2s
+    Open New Notebook
+    ${commit_msg2}=    Get Last Commit Message
+    Should Not Be Equal    ${commit_msg2}    ${commit_msg1}
+    Clean Up Server
+
+
+*** Keywords ***
+Server Setup
+    [Documentation]    Suite Setup
+    Begin Web Test
+    Launch JupyterHub Spawner From Dashboard
+    Spawn Notebook With Arguments    image=s2i-minimal-notebook    size=Default
+
+Push Some Changes to Repo
+    [Arguments]    ${github username}    ${token}    ${filepath}    ${githublink}    ${commitmsgg}
```

Comment:
Remember `[Documentation]` for all keywords

Reply:
Done